docs/proposal/controller/internal-api.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package controller;
+import "google/protobuf/empty.proto";
+
+//Represents the liveness status of 
+enum Status {
+  RUNNING = 0;
+  STARTING = 1;
+  STOPPED = 2;
+  STOPPING = 3;
+  DESTROYING = 4;
+  TERMINATED = 5;
+  CRASHED = 6;
+  FAILED = 7;
+  SCHEDULING = 8;
+  SCHEDULED = 9;
+}
+
+//Represents the type of workload running
+enum Type {
+  CONTAINER = 0;
+}
+
+// Represents a Instance status message
+message InstanceStatus {
+  string id = 1;
+  Status status = 2;
+  string description = 3;
+}
+
+// Represents a Node status message
+message NodeStatus {
+  string id = 1;
+  Status status = 2;
+  string description = 3;
+  Resource resource = 4;
+  repeated Instance instances = 5;
+}
+
+//Represents an Instance running
+message Instance {
+  string id = 1;
+  string name = 2;
+  Type type = 3;
+  Status status = 4;
+  string uri = 5;
+  repeated string environnement = 6;
+  Resource resource = 7;
+  repeated string ports = 8;
+  string ip = 9;
+}
+
+//Prepresent a resource entry
+message ResourceSummary {
+  int32 cpu = 1;
+  int32 memory = 2;
+  int32 disk = 3;
+}
+
+//Represents the resources used & threshold
+message Resource {
+  ResourceSummary max = 1;
+  ResourceSummary usage = 2;
+}
+
+service SchedulerService {
+  rpc NodeStatusUpdate(NodeStatus) returns (google.protobuf.Empty) {}
+  rpc InstanceStatusUpdate(InstanceStatus) returns (google.protobuf.Empty) {}
```

Comment:
A little nitty: `NodeUpdate`and `InstanceUpdate`.

Reply:
We agreed on this naming convention but i think we will rename it `UpdateNodeStatus` & `UpdateInstanceStatus`.